xbyak/xbyak_error.h

```diff
@@ -0,0 +1,169 @@
+#pragma once
+#ifndef XBYAK_XBYAK_ERROR_H_
+#define XBYAK_XBYAK_ERROR_H_
+
+#include <assert.h>
+#include <stdlib.h>
+#include <exception>
+namespace Xbyak {
+
+enum {
+	ERR_NONE = 0,
+	ERR_BAD_ADDRESSING,
+	ERR_CODE_IS_TOO_BIG,
+	ERR_BAD_SCALE,
+	ERR_ESP_CANT_BE_INDEX,
+	ERR_BAD_COMBINATION,
+	ERR_BAD_SIZE_OF_REGISTER,
+	ERR_IMM_IS_TOO_BIG,
+	ERR_BAD_ALIGN,
+	ERR_LABEL_IS_REDEFINED,
+	ERR_LABEL_IS_TOO_FAR,
+	ERR_LABEL_IS_NOT_FOUND,
+	ERR_CODE_ISNOT_COPYABLE,
+	ERR_BAD_PARAMETER,
+	ERR_CANT_PROTECT,
+	ERR_CANT_USE_64BIT_DISP,
+	ERR_OFFSET_IS_TOO_BIG,
+	ERR_MEM_SIZE_IS_NOT_SPECIFIED,
+	ERR_BAD_MEM_SIZE,
+	ERR_BAD_ST_COMBINATION,
+	ERR_OVER_LOCAL_LABEL, // not used
+	ERR_UNDER_LOCAL_LABEL,
+	ERR_CANT_ALLOC,
+	ERR_ONLY_T_NEAR_IS_SUPPORTED_IN_AUTO_GROW,
+	ERR_BAD_PROTECT_MODE,
+	ERR_BAD_PNUM,
+	ERR_BAD_TNUM,
+	ERR_BAD_VSIB_ADDRESSING,
+	ERR_CANT_CONVERT,
+	ERR_LABEL_ISNOT_SET_BY_L,
+	ERR_LABEL_IS_ALREADY_SET_BY_L,
+	ERR_BAD_LABEL_STR,
+	ERR_MUNMAP,
+	ERR_OPMASK_IS_ALREADY_SET,
+	ERR_ROUNDING_IS_ALREADY_SET,
+	ERR_K0_IS_INVALID,
+	ERR_EVEX_IS_INVALID,
+	ERR_SAE_IS_INVALID,
+	ERR_ER_IS_INVALID,
+	ERR_INVALID_BROADCAST,
+	ERR_INVALID_OPMASK_WITH_MEMORY,
+	ERR_INVALID_ZERO,
+	ERR_INVALID_RIP_IN_AUTO_GROW,
+	ERR_INVALID_MIB_ADDRESS,
+	ERR_X2APIC_IS_NOT_SUPPORTED,
+	ERR_NOT_SUPPORTED,
+	ERR_INTERNAL // Put it at last.
+};
+
+#ifndef XBYAK_NOEXCEPT
+    #define XBYAK_SET_STATUS(err) throw Error(err)
+    #define XBYAK_GET_STATUS ERR_NONE
+#else
+    #if (__cplusplus >= 201103)
+        #define THREAD_LOCAL thread_local
+    #else
+        #if defined(WIN32)
+        #define THREAD_LOCAL __declspec(thread)
+        #else
+        #define THREAD_LOCAL __thread
+        #endif
+    #endif
+class Status {
+private:
+    int status_;
+    Status() {}
+#if (__cplusplus >= 201103)
+    Status& operator=(const Status& other) = delete;
+    Status(const Status& other) = delete;
+#else
+    Status& operator=(const Status& other);
+    Status(const Status& other);
+#endif
```

Comment:
Probably should be enough to just have:

```suggestion
    Status& operator=(const Status& other);
    Status(const Status& other);
```

Reply:
Thanks. I guess just having it private for the user should be enough.